### ISSUE TITLE
Features for version 8

### ISSUE THREAD

**Issue by pwelter34** [serilog-contrib/serilog-sinks-azuretablestorage#86]
Here are my target features for version 8

- [x] Reduce code duplication from AzureBatchingTableStorageSink and AzureBatchingTableStorageWithPropertiesSink
- [x] create new IDocumentFactory to customize the stored document.  user can inject a different version to have full control over store document
- [x] simplify extension methods for LoggerSinkConfiguration
- [x] explore async option
- [x] support document expiration, cleanup old log entries

**Comment by heroes3lover** [comment]
Good news. any news about what time will v7 be published?

**Comment by germansak** [comment]
Hi, I don't know if this is the right place since I'm not familiar with contributing to OSS projects. For a specific need in an implementation that I'm working on, I forked this repo and added a "removeColumns" of type string[] to remove Serilog default columns (like MessageTemplate, RenderedMessage and so on) and it's configured with the logger itself like this:

```csharp
configuration.AzureTableStorageWithProperties(azureTableStorageConnectionString,
  storageTableName: "BusinessLogs",
  removeColumns: new string[]
  {
      "MessageTemplate",
      "RenderedMessage",
      "Exception",
      "AggregatedProperties"
  });
```

If it's something that could be useful on this project I could make a pull request. If not, I'd like to know how to proceed with the modifications I've made. Can I use them or should I ask for permissions to somebody?

Thanks in advance and sorry for my english (it's not my native language)
